### ISSUE TITLE
Update contained_in_s field in traject for Alma

### ISSUE THREAD

**Issue by hackartisan** [pulibrary/marc_liberation#922]
https://github.com/pulibrary/marc_liberation/blob/a691bf10e8f475172a586921d003f8f280329589/marc_to_solr/lib/traject_config.rb#L536-L541

**Comment by christinach** [comment]
Checking the 773 w of some alma records I realized that it carries the voyager id. I'm opening a new issue for an alma data cleanup. @mzelesky https://github.com/pulibrary/marc_liberation/issues/950

**Comment by christinach** [comment]
@mzelesky I started looking records that have 774 and 773 but I need more input on why we want `if there is a 773 we need to retrieve the holding and item information from that linked record.` part of the comments we added in the traject.

**Comment by mzelesky** [comment]
This is related to bound-withs. See https://knowledge.exlibrisgroup.com/Alma/Implementation_and_Migration/Migration_Guides_and_Tutorials/Voyager_to_Alma_Migration_Guide#Boundwiths for more information. This documentation covers how Voyager boundwiths are being migrated. I do not yet know how new boundwiths behave; I will look for this documentation.

After the new Alma environment is delivered on January 4th, we will have some examples.

**Comment by escowles** [comment]
Here's is Don's proposed solution for bound withs: https://docs.google.com/document/d/16CZuf0CzCxPSSD06mXOldKotdwwQ-K0VJiB12MhlQcE/edit

TL;DR: Creating a host record that links to the parts with 774, and the parts link to the host with 773.

**Comment by mzelesky** [comment]
This is a working model for how to retrieve availability and requesting for a bound-with record (more technically, a constituent record).

1. Retrieve availability of the constituent record (the record the user is looking at) using retrieve bib Alma API.
2. Retrieve availability of the MMS ID(s) in the 773 |w fields.
3. Combine both availability responses in the display.

**Comment by christinach** [comment]
The question here is if we are going to add any logic in the traject file during indexing to get the location or if we are going to do this on the fly when we load the record page for the constituent record.

**Comment by hackartisan** [comment]
9962646063506421 could be used as a fixture for this process and more can be created if needed.

**Comment by hackartisan** [comment]
For go-live we will only worry about indexing the MMS ids of the host record in this field. Note there may be multiple host ids. (fixture needed)

For go-live we will only add host holding during availability calls. It will mean these records can't be found during an advanced search by location or faceting by location. This is a small service degradation and acceptable until we develop a plan for indexing.

Talking to @mzelesky
- To do this during a full index would require 300k extra API calls.

Brainstorm of alternatives:
- Do this during a 2nd pass on indexing where you find all the constituent records and add the holdings from their host records. This would require investigating turning on partial updates for solr.
- If we could know that we always have the host records indexed first, then we could pull the holdings from them each time we hit a constituent record.
  - We could manage 2 sets of dumps: (non-bound-with and host records) and constituent records; or host records and (non-bound-with and constituent records).

**Comment by christinach** [comment]
- To move on with resolving this issue do I want only to index the host mms_ids 773 | w (maybe more than one)?
- Do we want to open a different issue for the availability api calls using the mms_ids from the 773 | w? Or should I handle it in this issue?

**Comment by christinach** [comment]
I will find a fixture.

**Comment by hackartisan** [comment]
Anything related to the availability calls should be part of #573. I will add a note there.
